pkg/syncer/cnsoperator/controller/cnsregistervolume/cnsregistervolume_controller.go

```diff
@@ -505,6 +512,112 @@ func (r *ReconcileCnsRegisterVolume) Reconcile(ctx context.Context,
+					if storagePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage != nil {
+						patchedStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Add(
+							*resource.NewQuantity(capacity.Value(), capacity.Format))
+					} else {
+						var (
+							usedQty     resource.Quantity
+							reservedQty resource.Quantity
+						)
+						reservedQty = *resource.NewQuantity(capacity.Value(), capacity.Format)
+						patchedStoragePolicyUsageCR.Status = storagepolicyusagev1alpha1.StoragePolicyUsageStatus{
+							ResourceTypeLevelQuotaUsage: &storagepolicyusagev1alpha1.QuotaUsageDetails{
+								Reserved: &reservedQty,
+								Used:     &usedQty,
+							},
+						}
+					}
+					err := patchStoragePolicyUsage(ctx, cnsOperatorClient, storagePolicyUsageCR,
+						patchedStoragePolicyUsageCR)
+					if err != nil {
+						log.Errorf("patching operation failed for StoragePolicyUsage CR: %q in namespace: %q. err: %v",
+							storagePolicyUsageCR.Name, storagePolicyUsageCR.Namespace, err)
+					}
+					// Retrieve the CR
+					currentStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+					finalStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+					key := apitypes.NamespacedName{Namespace: storagePolicyUsageCR.Namespace,
+						Name: storagePolicyUsageCR.Name}
+					err = cnsOperatorClient.Get(ctx, key, currentStoragePolicyUsageCR)
+					if err != nil {
+						log.Errorf("failed to get %s CR from supervisor namespace %q. Error: %+v",
+							storagePolicyUsageCR.Name, storagePolicyUsageCR.Namespace, err)
+						return reconcile.Result{RequeueAfter: timeout}, nil
+					}
+					finalStoragePolicyUsageCR = currentStoragePolicyUsageCR.DeepCopy()
+					// Decrease the Reserved field for StoragePolicyUsageCR
+					finalStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Sub(
+						*resource.NewQuantity(currentStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Value(),
+							currentStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Format))
+					// Increase the Used field for StoragePolicyUsageCR
+					finalStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Used.Add(
+						*resource.NewQuantity(currentStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Value(),
+							currentStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Format))
+					err = patchStoragePolicyUsage(ctx, cnsOperatorClient, currentStoragePolicyUsageCR,
+						finalStoragePolicyUsageCR)
+					if err != nil {
+						log.Errorf("patching operation failed for StoragePolicyUsage CR: %q in namespace: %q. err: %v",
+							currentStoragePolicyUsageCR.Name, currentStoragePolicyUsageCR.Namespace, err)
+					} else {
+						log.Infof("Successfully increased the used field by %v Mb "+
+							"for storagepolicyusage CR: %q in namespace: %q",
+							int64(pv.Spec.Capacity.Storage().ScaledValue(resource.Mega)), finalStoragePolicyUsageCR.Name,
```

Comment:
Can you also add message about decreasing the Reserved field in the log?

Reply:
Done